### PR TITLE
Use Stream-Style Serialization in Container Registry

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Updates `com.azure:azure-containers-containerregistry` to use stream-style serialization instead of Jackson Databind.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by lmolkova — COMMENTED** [review]
Looks great! And huge thanks for optimizing swagger!

I just have a concern about GA timeline.

File: sdk/containerregistry/azure-containers-containerregistry/pom.xml [hunk 1]
```diff
@@ -46,6 +46,11 @@
       <groupId>com.azure</groupId>
       <artifactId>azure-core</artifactId>
       <version>1.36.0</version> <!-- {x-version-update;com.azure:azure-core;dependency} -->
+    </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-json</artifactId>
+      <version>1.0.0-beta.2</version> <!-- {x-version-update;com.azure:azure-json;dependency} -->
```
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/containerregistry/azure-containers-containerregistry/pom.xml:53]: I think that should be doable, I'll work on hardening azure-json, and implementations. Worst case, we move this work to a feature branch and unblock main for a path forward to GA in April.

[the remark this answers is not in the document]
Comment (lmolkova) [reply sdk/containerregistry/azure-containers-containerregistry/pom.xml:53]: sounds good - let's try  :)

**Review by lmolkova — APPROVED** [review]
🎉

**Pull request merged**